modules/blockbuilder/partition_writer.go

```diff
@@ -18,7 +18,7 @@ import (
 )
 
 type partitionSectionWriter interface {
-	pushBytes(tenant string, req *tempopb.PushBytesRequest) error
+	pushBytes(tenant string, req *tempopb.PushBytesRequest, startTime time.Time) error
```

Comment:
`startTime` can be a field of the `writer` since it doesn't change for each section.